src/plugins/mpi/pmix/mpi_pmix.c

```diff
@@ -122,6 +128,63 @@ static void *_libpmix_open(void)
 	return lib_plug;
 }
 
+/*
+ * thread for codes from abort
+ */
+static void *_pmix_abort_thread(void *unused)
+{
+	PMIXP_DEBUG("Start abort thread");
```

Comment:
* Need EIO engine to progress 2 possible connections
  * server FD for connections from stepd's
  * local PIPE fd with termination request.
* Use Slurm protocol API on the server side:
  * init msg engine [[link](https://github.com/BorisBochkaryov/slurm/blob/master/src/common/slurm_protocol_socket.c#L407)] to prepare listen FD
  * Use this FD to pass it to EIO for polling
  * use accept message stream [[link](https://github.com/BorisBochkaryov/slurm/blob/master/src/common/slurm_protocol_socket.c#L449)]
  * Use recv_from_timeout [[link](https://github.com/BorisBochkaryov/slurm/blob/master/src/common/slurm_protocol_socket.c#L132)] to receive message.